packages/ui/src/common/hooks/useLocalStorage.ts

```diff
@@ -41,11 +41,23 @@ export const useLocalStorage = <T>(key?: string) => {
     setState(getItem(key))
   }, [key])
 
+  useEffect(() => {
+    const handleEventOnce = (event: any) => {
+      setState(getItem(key))
+    }
+
+    document.addEventListener(`storage_event_${key}`, handleEventOnce)
+    return () => {
+      document.removeEventListener(`storage_event_${key}`, handleEventOnce)
+    }
+  }, [])
```

Comment:
I'm not sure if the key is ever changed somewhere in the code base, but to at least be consistent with other hooks dependencies called here:
```suggestion
  }, [key])
```